noether_tpp/include/noether_tpp/mesh_modifiers/upsampling_modifier.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <noether_tpp/core/mesh_modifier.h>
+#include <pcl/PolygonMesh.h>
+#include <pcl/point_types.h>
+#include <map>
+#include <vector>
+
+namespace noether
+{
+class UpsamplingMeshModifier : public MeshModifier
```

Comment:
Do you want to call this something a little more specific like `AreaBasedUpsamplingMeshModifier`? I could also see another implementation of this upsampler that looks at edge length rather than area which would need a differentiated class name

Reply:
We could also make another instance of this based on [PCL mesh subdivision using VTK](https://pointclouds.org/documentation/classpcl_1_1_mesh_subdivision_v_t_k.html)